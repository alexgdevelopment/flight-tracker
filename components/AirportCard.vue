<template>
  <v-card elevation="5">
    <v-list-item
      ><v-list-item-content
        ><v-list-item-title data-testid="airportName">{{
          airport.fullName
        }}</v-list-item-title
        ><v-list-item-subtitle
          >Airport name</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >
    <v-list-item
      ><v-list-item-content
        ><v-list-item-title data-testid="airportCountryName">{{
          airport.country.name
        }}</v-list-item-title
        ><v-list-item-subtitle
          >Country</v-list-item-subtitle
        ></v-list-item-content
      ></v-list-item
    >
  </v-card>
</template>

<script>
  export default {
    name: 'AirportCard',
    props: {
      airport: {
        type: Object,
        required: true,
      },
    },
  }
</script>