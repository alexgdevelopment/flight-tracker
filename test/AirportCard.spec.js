import { mount } from '@vue/test-utils'
import { mockAirport } from './mockObjects'
import AirportCard from '@/components/AirportCard.vue'

describe('Airport Card', () => {
  let wrapper = null

  beforeEach(() => {
    wrapper = mount(AirportCard, {
      propsData: { airport: mockAirport },
    })
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('has a v-card element', () => {
    expect(wrapper.find('.v-card')).toBeTruthy()
  })
})
