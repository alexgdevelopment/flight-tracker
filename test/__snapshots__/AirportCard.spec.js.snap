// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airport Card should match the snapshot 1`] = `
<div
  class="v-card v-sheet theme--light elevation-5"
>
  <div
    class="v-list-item theme--light"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <div
        class="v-list-item__title"
      >
        Mock Airport
      </div>
      <div
        class="v-list-item__subtitle"
      >
        Airport name
      </div>
    </div>
  </div>
   
  <div
    class="v-list-item theme--light"
    tabindex="-1"
  >
    <div
      class="v-list-item__content"
    >
      <div
        class="v-list-item__title"
      >
        Tadjikistan
      </div>
      <div
        class="v-list-item__subtitle"
      >
        Country
      </div>
    </div>
  </div>
</div>
`;
